megamek/src/megamek/client/ui/dialogs/EntityReadoutDialog.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2021 - The MegaMek Team. All Rights Reserved.
+ *
+ * This file is part of MegaMek.
+ *
+ * MegaMek is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * MegaMek is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MegaMek. If not, see <http://www.gnu.org/licenses/>.
+ */
+package megamek.client.ui.dialogs;
+
+import java.awt.Container;
+import java.util.Objects;
+import javax.swing.*;
+import megamek.client.ui.baseComponents.AbstractDialog;
+import megamek.client.ui.panes.EntityViewPane;
+import megamek.common.*;
+
+/** A dialog showing the unit readout for a given unit. */
+public class EntityReadoutDialog extends AbstractDialog {
+
+    private final JFrame parent;
+    private final Entity entity;
+    private EntityViewPane entityView;
+
+    /** Constructs a non-modal dialog showing the readout (TRO) of the given entity. */ 
+    public EntityReadoutDialog(final JFrame frame, final Entity entity) {
+        this(frame, false, entity);
+    }
+
+    /** Constructs a dialog showing the readout (TRO) of the given entity with the given modality. */
+    public EntityReadoutDialog(final JFrame frame, final boolean modal, final Entity entity) {
+        super(frame, modal, "EntityReadoutDialog", "EntityReadoutDialog.title");
+        setTitle(getTitle() + entity.getShortNameRaw());
+        this.entity = Objects.requireNonNull(entity);
+        parent = frame;
+        initialize();
+    }
+
+    @Override
+    protected Container createCenterPane() {
+        entityView = new EntityViewPane(parent, entity);
+        return entityView;
+    }
+    

```

Comment:
```suggestion
    
    @Override
    protected void setCustomPreferences(final PreferencesNode preferences) {
        super.setCustomPreferences(preferences);
        preferences.manage(new JTabbedPanePreference(entityView));
    }
```

Reply:
I'd leave the preferences off of the AbstractUnitSelectorDialog for now, as any implementation will likely cause issues with MekHQ.

Reply:
I don't know that this does, but I've added it.

Reply:
So, what this does is it creates the preference that now handles all of the backend of monitoring the tabbed pane. Pretty much a black box that you need to make a one-line call for to ensure the preferences are handled properly.